client/v2/client_test.go

```diff
@@ -257,16 +258,19 @@ func TestClient_Concurrent_Use(t *testing.T) {
 	wg.Add(3)
 	n := 1000
 
+	errC := make(chan error, 3)
```

Comment:
Rather than counting the number of times this gets used, I would suggest using an unbuffered channel and launch a separate goroutine that waits on the WaitGroup and closes the channel. The last few lines would then be:

```go
go func() {
    wg.Wait()
    close(errC)
}

for e := range errC {
    if e != nil {
        t.Error(e)
    }
}
```